app/controllers/accept_account_invitations_controller.rb

```diff
@@ -4,8 +4,15 @@ class AcceptAccountInvitationsController < ApplicationController
   def show
     return if user_signed_in?
 
-    session[:after_sign_in_url] = request.fullpath
-    redirect_to new_user_session_url(email: AccountInvitation.find_by!(token: ps.fetch(:token)).email)
+    session[:after_authentication_url] = request.fullpath
+
+    invitee_email = AccountInvitation.find_by!(token: ps[:token]).email
```

Comment:
```suggestion
    invitee_email = AccountInvitation.find_by!(token: ps.fetch(:token)).email
```

If the `ps.fetch(:token)` bothers you, create a private method `token_param` or something and use it in all places.